pkg/plugins/policies/core/xds/meshroute/listeners.go

```diff
@@ -58,7 +58,7 @@ func MakeHTTPSplit(
 
 type DestinationService struct {
 	Outbound    mesh_proto.OutboundInterface
-	Port        *uint32
+	Port        *uint32 // todo(jakubdyszkiewicz): move to TargetRef when we introduce port in TargetRef
```

Comment:
This might just be a clumsy structuring of this intermediate type on my part. I think we don't necessarily need TargetRef with a port.
I think the `TargetRef` in this struct should be a `BackendRef` instead, whose weight we ignore, similar to how we handle mirroring in MeshHTTPRoute. And we put the port there in the "real `MeshService`" case.